tests/extras/datasets/spark/test_deltatable_dataset.py

```diff
@@ -0,0 +1,41 @@
+import pytest
+from delta import DeltaTable
+from pyspark.sql import SparkSession
+from pyspark.sql.types import IntegerType, StringType, StructField, StructType
+
+from kedro.extras.datasets.spark import DeltaTableDataset, SparkDataSet
+
+
+@pytest.fixture(autouse=True)
+def delta_spark_session(spark_session):
+    return spark_session
+
+
+@pytest.fixture
+def sample_spark_df():
+    schema = StructType(
+        [
+            StructField("name", StringType(), True),
+            StructField("age", IntegerType(), True),
+        ]
+    )
+
+    data = [("Alex", 31), ("Bob", 12), ("Clarke", 65), ("Dave", 29)]
+
+    return SparkSession.builder.getOrCreate().createDataFrame(data, schema)
+
+
+class TestDeltaTableDataSet:
+    def test_load(self, tmp_path, sample_spark_df):
+        filepath = (tmp_path / "data.csv").as_posix()
+        spark_ds = SparkDataSet(filepath=filepath, file_format="csv")
+        spark_ds.save(sample_spark_df)
+
+        spark_delta_ds = SparkDataSet(
+            filepath=filepath, file_format="delta", save_args={"mode": "overwrite"}
+        )
+        spark_delta_ds.save(sample_spark_df)
+        delta_ds = DeltaTableDataset(filepath=filepath)
+        delta_table = delta_ds.load()
+
+        assert isinstance(delta_table, DeltaTable)
```

Comment:
So personally I think this is exactly what we're looking for.
I would do the following tests:

1. Test loading `dataframewriter_options` from typical yaml (or already parsed into dict) and checking how it resolves - you can see some of the options here https://docs.delta.io/latest/quick-start.html#read-older-versions-of-data-using-time-travel
2. The above + seeing how those options are set on the `spark.sql.DataFrameWriter` object (probs protected properties to check - see here https://spark.apache.org/docs/latest/api/python/_modules/pyspark/sql/readwriter.html#DataFrameWriter)
3. The obvious - `spark.SparkDataSet` with `file_format = "delta"` but incorrect `mode`